docs/source/contribute/documentation/examples/tabs.md

```diff
@@ -0,0 +1,224 @@
+    
+    Windows has no ``sed`` equivalent natively installed. Instead, in order to replace every instance of the word "Hello" with the word "World" in a file, use the following command in Powershell:
+    
+    .. code-block:: shell
+      
+      (Get-Content index.rst) -replace 'Hello', 'World' | Out-File -encoding ASCII index.rst
+
+````
+
+
+``````{code-tab} md
+
+`````{tabs}
+
+````{group-tab} MacOS (BSD sed)
+
+To replace every instance of the word "Hello" with the word "World" in a file using BSD sed, use the following command:
+
+```{code-block} shell
+sed -i '' 's/Hello/World/g' index.rst
+```
+````
+
+````{group-tab} Linux (GNU sed)
+
+To replace every instance of the word "Hello" with the word "World" in a file using GNU sed, use the following command:
+
+```{code-block} shell
+sed -i -e 's/README/index/g' index.rst
+```
+````
+
+````{group-tab} Windows (Powershell)
+
+Windows has no ``sed`` equivalent natively installed. Instead, in order to replace every instance of the word "Hello" with the word "World" in a file, use the following command in Powershell:
+
+```{code-block} shell
+(Get-Content index.rst) -replace 'Hello', 'World' | Out-File -encoding ASCII index.rst
+```
+````
+`````
+``````
+```````
+
+Renders as:
+
+`````{tabs}
+
+````{group-tab} MacOS (BSD sed)
+
+To replace every instance of the word "Hello" with the word "World" in a file using BSD sed, use the following command:
+
+```{code-block} shell
+sed -i '' 's/Hello/World/g' index.rst
+```
+````
+
+````{group-tab} Linux (GNU sed)
+
+To replace every instance of the word "Hello" with the word "World" in a file using GNU sed, use the following command:
+
+```{code-block} shell
+sed -i -e 's/README/index/g' index.rst
+```
+````
+
+````{group-tab} Windows (Powershell)
+
+Windows has no ``sed`` equivalent natively installed. Instead, in order to replace every instance of the word "Hello" with the word "World" in a file,
+use the following command in Powershell:
+
+```{code-block} shell
+(Get-Content index.rst) -replace 'Hello', 'World' | Out-File -encoding ASCII index.rst
+```
+````
+
+`````
+
+## Code Tabs
+
+When the only content of a tab will be a [code-block](./code-blocks.md), use `code-tab` instead of `group-tab`. 
+
+### Syntax
+
+``````{tabs}
+```{code-tab} rst
+.. code-tab:: language (Optional Displayed Name)
+    
+  ...code...
+```
+
+````{code-tab} md 
+
+```{code-tab} language (Optional Displayed Name)
+...code...
+```
+````
+``````
+
+If you exclude the Optional Displayed Name, then the proper name of the given language will be displayed.
+
+### Usage
+
+``````{tabs}
+```{code-tab} rst
+
+.. tabs::
+  
+  .. code-tab:: shell MacOS (BSD sed)
+    
+    sed -i '' 's/Hello/World/g' index.rst
+
+  .. code-tab:: shell Linux (GNU sed)
+    
+    sed -i -e 's/README/index/g' index.rst
+
+  .. code-tab:: shell Windows (Powershell)
+    
+    (Get-Content index.rst) -replace 'Hello', 'World' | Out-File -encoding ASCII index.rst
+
+```
+
+`````{code-tab} md
```

Comment:
Just an observation - it seems like the tabs are coupled to each other (i.e. selecting markdown on one tab changes all the tabs on the page to be markdown). That's not necessarily bad, but I found it surprising.

https://user-images.githubusercontent.com/1189602/203616475-e8a1292c-734d-4195-a289-26b394eb6094.mov

Reply:
Yup, that's intentional behavior! `code-tab`s are a special type of `group-tab` that only contains a code-block. [There is the `tab` option as well](https://sphinx-tabs.readthedocs.io/en/latest/#basic-tabs), but I couldn't think of when we'd want to use that instead of `group-tab` (since we generally use `tabs` for API vs UI or RST vs MD documentation, where a user's selection ought to carry throughout the page), so I excluded mentioning it.